pkg/monitoring/domainstats/downwardmetrics/hostmetrics.go

```diff
@@ -77,10 +123,59 @@ func (h *hostMetricsCollector) Collect() (metrics []api.Metric) {
 
 func defaultHostMetricsCollector() *hostMetricsCollector {
 	return &hostMetricsCollector{
-		procCPUInfo: "/proc/cpuinfo",
-		procStat:    "/proc/stat",
-		procMemInfo: "/proc/meminfo",
-		procVMStat:  "/proc/vmstat",
-		pageSize:    os.Getpagesize(),
+		procPath: "/proc",
+		sysPath:  "/sys",
+		pageSize: os.Getpagesize(),
+	}
+}
+
+func derefOrZero(val *uint64) uint64 {
+	if val == nil {
+		return 0
+	}
+	return *val
+}
+
+type vmStat struct {
+	pswpin  uint64
+	pswpout uint64
+}
+
+// readVMStat reads specific fields from the /proc/vmstat file.
+// We implement it here, because it is not implemented in "github.com/prometheus/procfs"
+// library.
+func readVMStat(path string) (*vmStat, error) {
```

Comment:
nit: imo this function should, at least, live in its own file, where the type declaration can also be moved too